src/backend/distributed/transaction/transaction_management.c

```diff
@@ -213,6 +214,7 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			XactModificationLevel = XACT_MODIFICATION_NONE;
 			dlist_init(&InProgressTransactions);
 			CoordinatedTransactionUses2PC = false;
+			SubPlanLevel = 0;
```

Comment:
We've not done a good job of describing what we're doing in this function. Could you add a comment here saying what `SubPlanLevel` is and why we need to reset it?